### ISSUE TITLE
Issue documenting multiple mixin classes with virtual inheritance

### ISSUE THREAD

**Issue by japence** [doxygen/doxygen#10717]
I'm having issues properly documenting templated mixin classes that use virtual inheritance. As an example, I have an abstract interface class. The implementation of the abstract interface class is spread across multiple mixins, which inherit the abstract interface virtually. A concrete class inherits multiple mixins, templated on the abstract interface class.
```
class AbstractInterface {
public:
   virtual void funcA() = 0;
   virtual void funcB() = 0;
   void funcC() {}
};

template<class... Inf>
class MixinA : public virtual Inf... {
public:
   void funcA() override {}
};

template<class... Inf>
class MixinB : public virtual Inf... {
public:
   void funcB() override {}
};

class ConcreteClass : 
   public MixinA<AbstractInterface>,
   public MixinB<AbstractInterface>
{ };
```

**Expected behavior**
I would expect this to produce something that resembles a traditional diamond graph for a virtual inheritance hierarchy:
```
                  AbstractInterface
                      /    \
MixinA<AbstractInterface> MixinB<AbstractInterface>
                      \    /
                   ConcreteClass
```
I would also expect members of `AbstractInterface` to be displayed in the member documentation of `ConcreteClass` (e.g. `AbstractInterface::funcC()`).

**Actual Behavior**
What I'm seeing is a bit confusing. The template parameter name `Inf` is the root class. One of the inherited mixin classes shows the template type that was used, with a dashed yellow arrow to the mixin class template, and a solid blue arrow to the template parameter name `Inf`. All other inherited mixins just show the class template, connected to the template parameter name.
```
                     Inf
                    / | \
         MixinA<Inf>  |  \
                  \   |   \
<AbstractInterface>\  |    \
                    \ |     \
MixinA<AbstractInterface> MixinB<Inf>
                      \     /
                   ConcreteClass
```
There is no link between `ConcreteClass` and `AbstractInterface` in any of the class hierarchy or collaboration graphs, and no documentation of members inherited from just the abstract interface class (e.g. `AbstractInterface::funcC()`) in the `ConcreteClass` documentation.

I haven't been able to produce correct documentation using `@implements` and `@extends` commands in my comments either.

**Version**
1.10.0

**Comment by japence** [comment]
My description of the actual behavior isn't completely accurate. I only see that unbalanced inheritance diagram when the Mixin classes are scoped within different namespaces (e.g. `Foo::Bar::Mixin`, more on that later). Without the namespaces I get a balanced graph, but there's still no connection from `ConcreteClass` to `AbstractClass` and none of the members of `AbstractClass` that haven't been overridden appear in `ConcreteClass`.

**Source**
```
#pragma once
#include <iostream>

/** @brief An abstract interface class */
class AbstractInterface {
public:
   virtual int funcA() = 0;
   virtual int funcB() = 0;
   int sum() { return funcA() + funcB(); }
};

/** @brief Mixin that implements funcA() */
template<class... Inf>
class MixinA : public virtual Inf... {
public:
   int funcA() override { return 1; }
};

/** @brief Mixin that implements funcB() */
template<class... Inf>
class MixinB : public virtual Inf... {
public:
   int funcB() override { return 2; }
};

/** @brief Implements AbstractInterface using MixinA and MixinB */
class ConcreteClass : 
   public MixinA<AbstractInterface>,
   public MixinB<AbstractInterface>
{ 
   ConcreteClass() { std::cout << sum(); }
};
```

**Screenshots**
![MixinDiagram](https://github.com/doxygen/doxygen/assets/103002728/5e860037-9559-4ed3-a8b4-b51dbff35548)
![MixinMembers](https://github.com/doxygen/doxygen/assets/103002728/c4bed244-ddf2-4830-9338-d96398cb74a2)
*Note that `sum()` doesn't appear as a public member of `ConcreteClass`.*

**Mixins from different namespaces**
 When the mixin classes are placed in different namespaces I get a diagram that looks like this:
![MixinDiagram2](https://github.com/doxygen/doxygen/assets/103002728/7116f3fc-06fc-47c4-89bd-ac950ef74edf)
And there's still no connection from `ConcreteClass` to `AbstractClass`

**Source**
```
#pragma once
#include <iostream>

/** @brief An abstract interface class */
class AbstractInterface {
public:
   virtual int funcA() = 0;
   virtual int funcB() = 0;
   int sum() { return funcA() + funcB(); }
};

namespace Foo {
   /** @brief Mixin that implements funcA() */
   template<class... Inf>
   class Mixin : public virtual Inf... {
   public:
      int funcA() override { return 1; }
   };
}

namespace Bar {
   /** @brief Mixin that implements funcB() */
   template<class... Inf>
   class Mixin : public virtual Inf... {
   public:
      int funcB() override { return 2; }
   };
}

/** @brief Implements AbstractInterface using Foo::Mixin and Bar::Mixin */
class ConcreteClass : 
   public Foo::Mixin<AbstractInterface>,
   public Bar::Mixin<AbstractInterface> {
public:
   ConcreteClass() { std::cout << sum(); }
};
```

**Doxyfile**
[Doxyfile.zip](https://github.com/doxygen/doxygen/files/14541543/Doxyfile.zip)

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.11.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
